var/spack/repos/builtin/packages/py-metomi-isodatetime/package.py

```diff
@@ -0,0 +1,19 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class PyMetomiIsodatetime(PythonPackage):
+    """Python ISO 8601 date time parser and data model/manipulation utilities."""
+
+    homepage = "https://github.com/metomi/isodatetime"
+    pypi = "metomi-isodatetime/metomi-isodatetime-1!3.0.0.tar.gz"
```

Comment:
Does this actually work, or does it replace `1!3.0.0` with `3.0.0`? We might need a `url_for_version` function. You can test this by running `spack fetch py-metomi-isodatetime` or `spack info py-metomi-isodatetime` to see what URL it uses.

Reply:
This works. It uses `https://files.pythonhosted.org/packages/source/m/metomi-isodatetime/metomi-isodatetime-1!3.0.0.tar.gz`